Automattic/vip
# Update eslint-plugin-wpvip to 0.5.8, Prettier updates

## Description

- Update `eslint-plugin-wpvip` to `0.5.8` which contains additional lint rules and expands Prettier config to work with JSON, Markdown, and YAML files.
- Install `wp-prettier`
- Add `.editorconfig`
- Standardize GitHub Action workflows to reduce unnecessary runs and run Prettier check.
- Standardize `npm` scripts
- Update code to fix lint errors
- Remove disabled rules in favor of ad hoc `eslint-disable-next-line`
- Apply Prettier fixes

### What should I review?

Since this is a formatting-heavy PR, I recommend reviewing commit-by-commit and ignoring the "Prettier formatting" commits. Code changes to address lint errors deserve attention, as do any changes to `package.json`, `.eslintrc`, and any changed ignore files.

It's not particularly helpful to review the formatting changes themselves, since they are automated and well-tested upstream.

## Steps to Test

1. Check out PR.
1. Run `npm run format`
1. Run `npm run lint`
